bitmap-autobots/index.js

```diff
@@ -0,0 +1,134 @@
+'use strict';
+
+const fs = require('fs');
+const Bitmap = require('./lib/bitmap.js');
+require('./lib/bitmap-transforms.js');
+
+let checkOutputDirectoryExists = function() {
+  return new Promise(function(resolve, reject) {
+    fs.exists('./output/', function(exists) {
+      if (exists) {
+        resolve();
+      }
+      else {
+        reject();
+      }
+    });
+  });
+};
+
+let createOutputDirectory = function() {
+  return new Promise(function(resolve, reject) {
+    fs.mkdir('./output', function(err) {
+      if (err) reject(err);
+      resolve();
+    });
+  });
+};
+
+let loadBitmap = function(filename) {
+  return new Promise(function(resolve, reject) {
+    fs.readFile(filename, function(err, buffer) {
+      if (err) reject(err);
+      resolve(new Bitmap(buffer));
+    });
+  });
+};
+
+let createOutput = function(fileName, bitmap, func) {
+  return new Promise(function(resolve, reject) {
+    let transformedBitmap = func();
+
+    fs.writeFile(`./output/${fileName}.bmp`, transformedBitmap.buffer, 'binary', function(err) {
+      if (err) reject(err);
+      resolve(bitmap);
+    });
+  });
+};
+
+let filePath = process.argv[2];
+let transforms = process.argv.slice(3);
+
+if (filePath && transforms) {
+  checkOutputDirectoryExists()
+    .catch(createOutputDirectory)
+    .then(() => loadBitmap(filePath))
+    .then(bitmap => {
+      while (transforms.length > 0) {
+        let transform = transforms.shift();
+        let transformParts = transform.split(':');
+
```

Comment:
while this switch statement works for what you are accomplishing another way to handle this that would only take up a single line of code and would keep it from growing out of control if you had 100+ transformations would be to access the property of the object. So instead of passing in a string of bw or hflip you could pass in the name of the property and just access the bitmap[property](). For the items with parameters you can have another variable you pass in to bitmap[property](param) and on the ones that don't need the parameter you can ignore it as they won't accept a parameter anyway.